src/index.js

```diff
@@ -57,21 +57,27 @@ export function checkSelector(selector) {
 
 
   const { dependencies = [], selectorName = null } = selector
+
   const isNamed = typeof selectorName === 'string'
   const recomputations = selector.recomputations ? selector.recomputations() : null
 
   const ret = { dependencies, recomputations, isNamed, selectorName }
   if (_getState) {
+    const extra = {}
     const state = _getState()
-    const inputs = dependencies.map((parentSelector) => parentSelector(state))
-    const extra = { inputs }
+
     try {
```

Comment:
So the subtlety here is the deeply nested try blocks which are a bit ugly. Presumably you don't want to try to calculate output if the inputs throw. I think that's consonant with the behavior of a typical ```reselect user```

Reply:
yes, if we will have an error in dependencies that we will have error in selector and execute the selector is unnecessarily in this case.
this changes needed for using the deep selectors with props